src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/AvoidSingleUseOfLocalJsonSerializerOptionsTests.cs

```diff
@@ -0,0 +1,574 @@
+
+class Program
+{{
+    private JsonSerializerOptions Options {{ get; set; }}
+
+    string Serialize<T>(T value)
+    {{
+        {snippet}
+        return JsonSerializer.Serialize(value, opt);
+    }}
+}}
+";
+            return test.RunAsync();
+        }
+
+        public static IEnumerable<object[]> CS_UseNewLocalOptionsAsArgument_FieldAssignment_NoWarn_TheoryData()
+        {
+            return CS_UseNewLocalOptionsAsArgument_Assignment_TheoryData(useField: true).Select(e => new object[] { e });
+        }
+
+        public static IEnumerable<object[]> CS_UseNewLocalOptionsAsArgument_PropertyAssignment_NoWarn_TheoryData()
+        {
+            return CS_UseNewLocalOptionsAsArgument_Assignment_TheoryData(useField: false).Select(e => new object[] { e });
+        }
+
+        private static List<string> CS_UseNewLocalOptionsAsArgument_Assignment_TheoryData(bool useField)
+        {
+            string target = useField ? "s_options" : "Options";
+
+            return new List<string>()
+            {
+                $@"JsonSerializerOptions opt = new JsonSerializerOptions();
+                    {target} = opt;",
+
+                $@"JsonSerializerOptions opt;
+                    {target} = opt = new JsonSerializerOptions();",
+
+                $@"JsonSerializerOptions opt = {target} = new JsonSerializerOptions();",
+
+                $@"JsonSerializerOptions opt = {target} ??= new JsonSerializerOptions();",
+
+                $@"JsonSerializerOptions opt = new JsonSerializerOptions();
+                    {target} ??= opt;",
+
+                $@"JsonSerializerOptions opt = new JsonSerializerOptions();
+                    ({target}, _) = (opt, 42);",
+
+                $@"JsonSerializerOptions opt = new JsonSerializerOptions();
+                    (({target}, _), _) = ((opt, 42), 42);"
+            };
+        }
+
+        [Fact]
+        public Task CS_UseNewLocalOptionsAsArgument_NotSingleUse_NoWarn()
+            => VerifyCS.VerifyAnalyzerAsync(@"
+using System.Text.Json;
+
+class Program
+{
+    static string SerializeTwice<T>(T value)
+    {
+        JsonSerializerOptions opt = new JsonSerializerOptions();
+
+        string str1 = JsonSerializer.Serialize(value, opt);
+        string str2 = JsonSerializer.Serialize(value, opt);
+
+        return str1 + str2;
+    }
+}
+");
+
+        [Theory]
+        [InlineData("opt1", "opt2")]
+        [InlineData("opt1", "opt3")]
+        [InlineData("opt2", "opt3")]
+        public Task CS_UseNewLocalOptionsAsArgument_MultiAssignment_NotSingleUse_NoWarn(string expression1, string expression2)
+            => VerifyCS.VerifyAnalyzerAsync($@"
+using System.Text.Json;
+
+class Program
+{{
+    static string Serialize<T>(T value)
+    {{
+        JsonSerializerOptions opt1, opt2, opt3;
+        opt1 = opt2 = opt3 = new JsonSerializerOptions();
+
+        string json1 = JsonSerializer.Serialize(value, {expression1});
+        string json2 = JsonSerializer.Serialize(value, {expression2});
```

Comment:
Its look like single use to me, for each instance

Reply:
it's not a single-use reference-wise. `opt1 = opt2 = opt3 = new JsonSerializerOptions();` does not create multiple instances.